### PR TITLE
mssql: support rails 7

### PR DESCRIPTION
**Pull request by stackmystack**

### CODE REVIEW

File: test/arelx_test_helper.rb [hunk 1]
```diff
@@ -22,26 +23,32 @@ def colored(color, msg)
 #     The issue also seems to be related to arel version: at some point, arel
 #     dropped its wide support for DBs and kept Postgres, MySQL and SQLite.
 #     Here, we're just trying to load the correct ones.
+#
+# NOTE:
+#     As of jruby 9.4 (and maybe 9.3, but I coldn't test it given the state of
```
Comment (akimd) [inline test/arelx_test_helper.rb:28]: couldn't

File: lib/arel_extensions/visitors/mssql.rb [hunk 2]
```diff
@@ -2,52 +2,68 @@ module ArelExtensions
...
-        'MM-DD-YYYY' => 110,
-        'YYYY/MM/DD' => 111,
-        'DD-MM-YYYY' => 105,
-        'DD-MM-YY'   => 5,
-        'DD.MM.YYYY' => 104,
-        'YYYY-MM-DDTHH:MM:SS:MMM' => 126
-      }.freeze
+      MSSQL_CLASS_NAMES = %i[MSSQL SQLServer].freeze
+
+      mssql_class =
```
Comment (akimd) [inline lib/arel_extensions/visitors/mssql.rb:7]: Why not a constant ?

**Review by akimd — APPROVED** [review]
LGTM.

Great piece of work!  Excellent news.

File: lib/arel_extensions/visitors/mssql.rb (hunk 2 above)
Comment (stackmystack) [reply lib/arel_extensions/visitors/mssql.rb:7]: Does it change a lot in this case? We're only going to use it for const_get later.

Comment (akimd) [reply lib/arel_extensions/visitors/mssql.rb:7]: If we don't intend to expose it in descendants, you are right it should stay a local var.

**Comment by jdelporte** [comment]
Everything here is done more or less for the same reason PR #108 was made, where literal date string (in sqlserver) are transformed into date object, Why is this revert ?

**Comment by jdelporte** [comment]
We should test on different mssql versions

**Comment by stackmystack** [comment]
> Everything here is done more or less for the same reason PR #108 was made, where literal date string (in sqlserver) are transformed into date object, Why is this revert ?

It's not technically a revert, because otherwise the tests from the PR would not pass.

Instead of formatting manually, we're using the `to_formatted_s`, which I picked up by reading arel itself. This allows us to align the patching we're doing with the original code, so maintaining those 2 copies becomes far less painful, as divergence can be eye-balled in a couple of seconds.

> We should test on different mssql versions

Absolutely agreed. I tried to add 2022 for instance, and that was a failure, so I left it as is.
I will have to look (yet again) for how to support more versions.

**Comment by jdelporte** [comment]
For the first point: Ok then.

For the versions, I was more thinking about older versions. But I guess we will need to run windows server images.

**Comment by stackmystack** [comment]
> For the versions, I was more thinking about older versions. But I guess we will need to run windows server images.

Yeah, MS doesn't provide old images, and it might be problematic that we do them ourselves. We might have to go (back) the travis route 😵‍💫.

**Pull request merged**